### PR TITLE
chore: add ESlint and pre commit

### PR DESCRIPTION
**Pull request by ChihYang41**
This PR does the following things:
1. add ESlint, Prettier, Husky, lint-staged 
2. format some files based on ESlint rules
3. delete tsconfig, prettierrc.js in example folder, it will apply the settings in root folder.

Please feel free to let me know if there is anything needs to be changed.

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -2,8 +2,10 @@ module.exports = {
...
   singleQuote: true,
-  tabWidth: 2,
   useTabs: false,
+  tabWidth: 2,
+  printWidth: 100,
+  jsxSingleQuote: false,
   trailingComma: 'all',
-  arrowParens: 'avoid',
-};
+  arrowParens: "avoid"
```
Comment (Dylan0916) [inline .prettierrc.js:10]: use single quote

File: .prettierrc.js [hunk 2]
```diff
@@ -2,8 +2,10 @@ module.exports = {
   semi: true,
   bracketSpacing: true,
   singleQuote: true,
-  tabWidth: 2,
   useTabs: false,
+  tabWidth: 2,
+  printWidth: 100,
```
Comment (Dylan0916) [inline .prettierrc.js:7]: I prefer 80

**Review by Dylan0916 — CHANGES_REQUESTED** [review]

File: .prettierrc.js (hunk 2 above)
Comment (ChihYang41) [reply .prettierrc.js]: okay 🙆‍♂️, I have changed it to `80`

File: .prettierrc.js (hunk 1 above)
Comment (ChihYang41) [reply .prettierrc.js]: fixed

**Review by Dylan0916 — APPROVED** [review]

**Pull request merged**